Source/WebCore/platform/gstreamer/GStreamerQuirkAmLogic.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2024 Igalia S.L
+ * Copyright (C) 2024 Metrological Group B.V.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "GStreamerQuirkAmLogic.h"
+
+#if USE(GSTREAMER)
+
+#include "GStreamerCommon.h"
+#include <wtf/OptionSet.h>
+
+namespace WebCore {
+
+GST_DEBUG_CATEGORY_STATIC(webkit_amlogic_quirks_debug);
+#define GST_CAT_DEFAULT webkit_amlogic_quirks_debug
+
+GStreamerQuirkAmLogic::GStreamerQuirkAmLogic()
+{
+    GST_DEBUG_CATEGORY_INIT(webkit_amlogic_quirks_debug, "webkitquirksamlogic", 0, "WebKit AmLogic Quirks");
+}
+
+GstElement* GStreamerQuirkAmLogic::createWebAudioSink()
+{
+    // autoaudiosink changes child element state to READY internally in auto detection phase
+    // that causes resource acquisition in some cases interrupting any playback already running.
+    // On Amlogic we need to set direct-mode=false prop before changing state to READY
+    // but this is not possible with autoaudiosink.
+    auto sink = makeGStreamerElement("amlhalasink", nullptr);
+    RELEASE_ASSERT_WITH_MESSAGE(sink, "amlhalasink should be available in the system but it is not");
+    g_object_set(sink, "direct-mode", FALSE, nullptr);
+    return sink;
+}
+
+bool GStreamerQuirkAmLogic::configureElement(GstElement* element, const OptionSet<ElementRuntimeCharacteristics>& characteristics)
+{
+    GST_INFO("Set property disable-xrun to TRUE");
+    g_object_set(element, "disable-xrun", TRUE, nullptr);
```

Comment:
This should check and configure only GstAmlHalAsink. Otherwise it produces a lot of noise:

```
Mar 19 20:33:27 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:27.940: g_object_set_is_valid_property: object class 'GstURIDecodeBin' has no property named 'disable-xrun'                                                                                                                   Mar 19 20:33:27 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:27.942: g_object_set_is_valid_property: object class 'WebKitWebSrc' has no property named 'disable-xrun'                                                                                                                      
Mar 19 20:33:27 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:27.942: g_object_set_is_valid_property: object class 'GstTypeFindElement' has no property named 'disable-xrun'                                                                                
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.307: g_object_set_is_valid_property: object class 'GstDecodeBin' has no property named 'disable-xrun'
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.307: g_object_set_is_valid_property: object class 'GstTypeFindElement' has no property named 'disable-xrun'                                                                                                                
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.307: g_object_set_is_valid_property: object class 'GstQueue2' has no property named 'disable-xrun'                                                                                                                         
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.334: g_object_set_is_valid_property: object class 'GstQTDemux' has no property named 'disable-xrun'                                                                                                                        
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.337: g_object_set_is_valid_property: object class 'GstMultiQueue' has no property named 'disable-xrun'                                                                                                                     
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.339: g_object_set_is_valid_property: object class 'GstH264Parse' has no property named 'disable-xrun'                                                                                                                      
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.341: g_object_set_is_valid_property: object class 'GstCapsFilter' has no property named 'disable-xrun'                                                                                                                     
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.344: g_object_set_is_valid_property: object class 'avdec_h264' has no property named 'disable-xrun'                                                                                                                        
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.349: g_object_set_is_valid_property: object class 'GstAacParse' has no property named 'disable-xrun'                                                                                                                       
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.374: g_object_set_is_valid_property: object class 'avdec_aac' has no property named 'disable-xrun'                                                                                                                  
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.401: g_object_set_is_valid_property: object class 'GstInputSelector' has no property named 'disable-xrun'                                                                                                                  
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.402: g_object_set_is_valid_property: object class 'GstInputSelector' has no property named 'disable-xrun'                                                                                                                  Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.403: g_object_set_is_valid_property: object class 'GstTee' has no property named 'disable-xrun'                                                                                                                            
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.406: g_object_set_is_valid_property: object class 'GstBin' has no property named 'disable-xrun'   
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.406: g_object_set_is_valid_property: object class 'GstQueue' has no property named 'disable-xrun'                                                                                                                          
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.406: g_object_set_is_valid_property: object class 'WebKitVideoSink' has no property named 'disable-xrun'                                                                                                                   
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.408: g_object_set_is_valid_property: object class 'GstBin' has no property named 'disable-xrun'                                                                                                                            
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.408: g_object_set_is_valid_property: object class 'GstPlaySinkAudioConvert' has no property named 'disable-xrun'                                                                                                           
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.409: g_object_set_is_valid_property: object class 'GstAudioResample' has no property named 'disable-xrun'                                                                                                                  
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.409: g_object_set_is_valid_property: object class 'GstAudioConvert' has no property named 'disable-xrun'                                                                                                                   
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.409: g_object_set_is_valid_property: object class 'GstIdentity' has no property named 'disable-xrun'                                                                                                                       
Mar 19 20:33:28 apache-4k HtmlApp-0[1707]: (WPEWebProcess:20): GLib-GObject-WARNING **: 16:33:28.409: g_object_set_is_valid_property: object class 'GstQueue' has no property named 'disable-xrun' 
```